game_logic.h

```diff
@@ -0,0 +1,21 @@
+#define BOARD_SIZE 4
+#define GOAL 3
+#define ALLOW_EXCEED 1
+#define N_GRIDS (BOARD_SIZE * BOARD_SIZE)
+#define GET_INDEX(i, j) ((i) * (BOARD_SIZE) + (j))
+#define GET_COL(x) ((x) % BOARD_SIZE)
+#define GET_ROW(x) ((x) / BOARD_SIZE)
+#define LOOKUP(table, i, j, else_value)                         \
```

Comment:
Move implementation specific macros such as `LOOKUP` out of the header file.

Reply:
I only moved `LOOKUP` because others are referenced by other program units.